pkg/oracles/contracts/StableLPOracle.sol

```diff
@@ -238,4 +252,46 @@ contract StableLPOracle is LPOracleBase {
     function _mulDownInt(int256 a, int256 b) internal pure returns (int256) {
         return (a * b) / _POSITIVE_ONE_INT;
     }
+
+    function _ensureValidPricesAndNormalize(
+        int256[] memory prices
+    ) internal view returns (int256[] memory normalizedPrices) {
+        if (prices[0] <= 0) {
+            revert InvalidOraclePrice();
+        }
+
+        int256 minPrice = prices[0];
+        int256 maxPrice = prices[0];
+        uint256 minPriceIndex = 0;
+
+        // Validate prices; find and validate bounds.
+        for (uint256 i = 1; i < _totalTokens; i++) {
+            if (prices[i] <= 0) {
+                revert InvalidOraclePrice();
+            }
+
+            if (prices[i] < minPrice) {
+                minPrice = prices[i];
+                minPriceIndex = i;
+            } else if (prices[i] > maxPrice) {
+                maxPrice = prices[i];
+            }
+        }
+
+        // The invariant of the pool gets distorted if the minimum price of the price feed array is too low.
+        if (minPrice < _MIN_PRICE_LIMIT) {
+            revert MinPriceTooLow();
+        }
+
+        // The `k` parameter has a high risk of not converging if the ratio between the maximum and minimum prices is
+        // too high.
+        if (maxPrice / minPrice > _PRICE_RATIO_LIMIT) {
+            revert PriceRatioTooHigh();
+        }
+
+        normalizedPrices = new int256[](_totalTokens);
+        for (uint256 i = 0; i < _totalTokens; i++) {
+            normalizedPrices[i] = i == minPriceIndex ? int256(FixedPoint.ONE) : _divDownInt(prices[i], minPrice);
```

Comment:
`_divDownInt(prices[i], minPrice);` should be exactly equal to `FixedPoint.ONE` for `i = minPriceIndex` anyway right? I don't think there's rounding error there (please double check and correct me if I'm wrong; we can even do a fuzz test if needed).

Reply:
And perhaps with that you don't even need to save and update the index.

Reply:
Yes, the divDown division would be `number * 1e18 / number`, which is 1e18 and has no rounding issues, no matter which number is used.

This is in place to save gas, as suggested by @EndymionJkb here: https://github.com/balancer/balancer-v3-monorepo/pull/1530#discussion_r2314640468 . But, maybe the cost of storing the index is greater than the division operation.

Reply:
Actually, @EndymionJkb optimization saves 200 gas.